### DISCUSSION TITLE
Winbar selector

### DISCUSSION THREAD

**Discussion by lucobellic** [folke/edgy.nvim#41] [Ideas]
Hello.

Could it be feasible to add winbar as selector similar to [neo-tree source selector](https://github.com/nvim-neo-tree/neo-tree.nvim#source-selector)?

In order to get multiple edgebar at the same position. Such as it might be possible to write something similar to this:
```lua
left = {
  {
    winbar_title = "Neo-tree",
    {  "neo-tree"  },
  },
  {
    winbar_title = "Search",
    {  "spectre_panel",   "Outline"  },
  },
}
```

Not sure if it could be easily added, but I feel that having both edgebar with winbar selector would be a killer feature.

Edit: added  video demonstration of what could be done (from [edgy-group.nvim](https://github.com/lucobellic/edgy-group.nvim)):

https://github.com/lucobellic/edgy-group.nvim/assets/6067072/6bee2762-9cc7-46d8-82a7-17b4e0f64cf7

**Comment by distek** [comment]
I made a thing:
https://github.com/distek/panel.nvim

I would love if this was just built in to edgy, though.

~One of the biggest issues I've come across with my plugin is things like trouble.nvim rely on the window ID it was opened with. I ended up forking trouble specifically to add a way to set the window ID, but it's hacky af and sometimes breaks in weird ways, so I wouldn't recommend using the fork.~

Edit to above: It appears that I've resolved the issue I was having with trouble.nvim.

The plugin stemmed from this same idea you bring up + using vertical windows in edgy being a pain to do things like auto-resize (and doing so just kinda felt meh to use)

**Reply by lucobellic** [reply]
Thanks for sharing your plugin, it decided me to put my personal configuration in a plugin.
[edgy-group.nvim](https://github.com/lucobellic/edgy-group.nvim) (for personal use and demonstration only)

It create groups of windows within edgebar by title and allow to close or open them with keymaps and commands.

I decided to use [bufferline custom_areas](https://github.com/akinsho/bufferline.nvim?tab=readme-ov-file#custom-areas) to create icons for the selected groups instead of integrating it directly into **edgy**.
The plugin has a lot of limitations but perfectly fits my personal needs if anyone is interested.

**Reply by distek** [reply]
Neat! Didn't even know that was a part of bufferline. Gunna look in to it. Thanks!
